app/controllers/concerns/pagination_concern.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module PaginationConcern
+  def extract_pagination_props
+    {
+      page: extract_int(params, :page),
+      limit: extract_int(params, :limit, 10)
+    }
+  end
+
+  def extract_int(target, key, default = 1)
+    value = target[key]
+
```

Comment:
why do you need to pass both of `params` and `:page` if you can pass the value instead?
```suggestion
  def extract_pagination_props
    {
      page: extract_int(params[:page]),
      limit: extract_int(params[:limit], 10)
    }
  end

  def extract_int(value, default = 1)
```